NegativeArraySizeException at net.openchrom.xxd.process.supplier.templates.ui

```
java.util.concurrent.ExecutionException: java.lang.NegativeArraySizeException
                at java.util.concurrent.FutureTask.report(FutureTask.java:122)
                at java.util.concurrent.FutureTask.get(FutureTask.java:192)
                at org.eclipse.chemclipse.support.ui.workbench.DisplayUtils.executeInUserInterfaceThread(DisplayUtils.java:103)
                at org.eclipse.chemclipse.support.ui.workbench.DisplayUtils.executeInUserInterfaceThread(DisplayUtils.java:82)
                at net.openchrom.xxd.process.supplier.templates.ui.core.PeakReviewDirectMSD.runProcess(PeakReviewDirectMSD.java:94)
                at net.openchrom.xxd.process.supplier.templates.ui.core.PeakReviewDirectMSD.identify(PeakReviewDirectMSD.java:54)
                at org.eclipse.chemclipse.chromatogram.msd.identifier.peak.PeakIdentifierMSD.identify(PeakIdentifierMSD.java:186)
                at org.eclipse.chemclipse.chromatogram.msd.identifier.peak.PeakIdentifierMSD.identify(PeakIdentifierMSD.java:215)
                at org.eclipse.chemclipse.chromatogram.msd.identifier.peak.PeakIdentifierMSD.identify(PeakIdentifierMSD.java:209)
                at org.eclipse.chemclipse.chromatogram.msd.identifier.peak.PeakIdentifierMSDProcessTypeSupplier$PeakIdentifierProcessorSupplier.apply(PeakIdentifierMSDProcessTypeSupplier.java:74)
                at org.eclipse.chemclipse.chromatogram.msd.identifier.peak.PeakIdentifierMSDProcessTypeSupplier$PeakIdentifierProcessorSupplier.apply(PeakIdentifierMSDProcessTypeSupplier.java:1)
                at org.eclipse.chemclipse.model.supplier.ChromatogramSelectionProcessorSupplier.apply(ChromatogramSelectionProcessorSupplier.java:39)
                at org.eclipse.chemclipse.model.supplier.IChromatogramSelectionProcessSupplier$1.execute(IChromatogramSelectionProcessSupplier.java:54)
                at org.eclipse.chemclipse.processing.supplier.IProcessSupplier.applyProcessor(IProcessSupplier.java:144)
                at org.eclipse.chemclipse.processing.methods.ProcessEntryContainer.applyProcessEntries(ProcessEntryContainer.java:190)
                at org.eclipse.chemclipse.processing.methods.ProcessEntryContainer.applyProcessEntries(ProcessEntryContainer.java:130)
                at org.eclipse.chemclipse.chromatogram.xxd.process.supplier.batchprocess.core.BatchProcess.execute(BatchProcess.java:69)
                at org.eclipse.chemclipse.chromatogram.xxd.process.supplier.batchprocess.ui.editors.BatchJobEditor.run(BatchJobEditor.java:190)
                at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:119)
Caused by: java.lang.NegativeArraySizeException
                at org.eclipse.chemclipse.ux.extension.xxd.ui.support.charts.ChromatogramChartSupport.getLineSeriesData(ChromatogramChartSupport.java:274)
                at org.eclipse.chemclipse.ux.extension.xxd.ui.support.charts.ChromatogramChartSupport.getLineSeriesData(ChromatogramChartSupport.java:175)
                at org.eclipse.chemclipse.ux.extension.xxd.ui.swt.PeakTracesUI.extractSIC(PeakTracesUI.java:148)
                at org.eclipse.chemclipse.ux.extension.xxd.ui.swt.PeakTracesUI.updateChart(PeakTracesUI.java:117)
                at org.eclipse.chemclipse.ux.extension.xxd.ui.swt.PeakTracesUI.setInput(PeakTracesUI.java:105)
                at org.eclipse.chemclipse.ux.extension.xxd.ui.swt.ExtendedPeakTracesUI.updatePeak(ExtendedPeakTracesUI.java:478)
                at org.eclipse.chemclipse.ux.extension.xxd.ui.swt.ExtendedPeakTracesUI.update(ExtendedPeakTracesUI.java:115)
                at net.openchrom.xxd.process.supplier.templates.ui.swt.peaks.ReviewController.update(ReviewController.java:133)
                at net.openchrom.xxd.process.supplier.templates.ui.swt.peaks.ExtendedTargetsUI.updateSelection(ExtendedTargetsUI.java:433)
                at net.openchrom.xxd.process.supplier.templates.ui.swt.peaks.ExtendedTargetsUI.setInput(ExtendedTargetsUI.java:102)
                at net.openchrom.xxd.process.supplier.templates.ui.swt.peaks.ReviewController.updateExtendedTargetsUI(ReviewController.java:381)
                at net.openchrom.xxd.process.supplier.templates.ui.swt.peaks.ReviewController.update(ReviewController.java:113)
                at net.openchrom.xxd.process.supplier.templates.ui.swt.peaks.ExtendedPeakReviewUI.updateSelection(ExtendedPeakReviewUI.java:364)
                at net.openchrom.xxd.process.supplier.templates.ui.swt.peaks.ExtendedPeakReviewUI.setInput(ExtendedPeakReviewUI.java:77)
                at net.openchrom.xxd.process.supplier.templates.ui.swt.peaks.ReviewController.updatePeakStatusUI(ReviewController.java:368)
                at net.openchrom.xxd.process.supplier.templates.ui.swt.peaks.ReviewController.update(ReviewController.java:99)
                at net.openchrom.xxd.process.supplier.templates.ui.swt.peaks.ReviewController.setInput(ReviewController.java:72)
                at net.openchrom.xxd.process.supplier.templates.ui.swt.peaks.PeakReviewControl.setInput(PeakReviewControl.java:40)
                at net.openchrom.xxd.process.supplier.templates.ui.swt.peaks.TemplatePeakReviewUI.setInput(TemplatePeakReviewUI.java:31)
                at net.openchrom.xxd.process.supplier.templates.ui.wizards.PeakReviewPage.createControl(PeakReviewPage.java:51)
                at org.eclipse.jface.wizard.Wizard.createPageControls(Wizard.java:176)
                at org.eclipse.jface.wizard.WizardDialog.createPageControls(WizardDialog.java:682)
                at org.eclipse.jface.wizard.WizardDialog.createContents(WizardDialog.java:574)
                at org.eclipse.jface.window.Window.create(Window.java:426)
                at org.eclipse.jface.dialogs.Dialog.create(Dialog.java:1096)
                at org.eclipse.jface.window.Window.open(Window.java:783)
                at net.openchrom.xxd.process.supplier.templates.ui.wizards.PeakReviewSupport.addSettings(PeakReviewSupport.java:53)
                at net.openchrom.xxd.process.supplier.templates.ui.core.PeakReviewDirectMSD$1.run(PeakReviewDirectMSD.java:101)
                at org.eclipse.chemclipse.support.ui.workbench.DisplayUtils$2.call(DisplayUtils.java:87)
                at org.eclipse.chemclipse.support.ui.workbench.DisplayUtils$2.call(DisplayUtils.java:1)
                at java.util.concurrent.FutureTask.run(FutureTask.java:266)
                at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:37)
                at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:182)
                at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:3906)
                at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3537)
                at org.eclipse.jface.operation.ModalContext$ModalContextThread.block(ModalContext.java:165)
                at org.eclipse.jface.operation.ModalContext.run(ModalContext.java:369)
                at org.eclipse.jface.dialogs.ProgressMonitorDialog.run(ProgressMonitorDialog.java:471)
                at org.eclipse.chemclipse.ux.extension.xxd.ui.swt.BatchJobUI$1.widgetSelected(BatchJobUI.java:200)
                at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:249)
                at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:86)
                at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4118)
                at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1052)
                at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:3931)
                at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3534)
                at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1170)
                at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
                at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1059)
                at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:153)
                at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:667)
                at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
                at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:597)
                at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
                at org.eclipse.chemclipse.rcp.app.ui.internal.support.ApplicationSupportDefault.start(ApplicationSupportDefault.java:26)
                at org.eclipse.chemclipse.rcp.app.ui.Application.start(Application.java:61)
                at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
                at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
                at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
                at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
                at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
                at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
                at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
                at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
                at java.lang.reflect.Method.invoke(Method.java:498)
                at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:656)
                at org.eclipse.equinox.launcher.Main.basicRun(Main.java:592)
                at org.eclipse.equinox.launcher.Main.run(Main.java:1498)
```

--- a/openchrom/plugins/net.openchrom.xxd.process.supplier.templates.ui/src/net/openchrom/xxd/process/supplier/templates/ui/core/PeakReviewDirectMSD.java
+++ b/openchrom/plugins/net.openchrom.xxd.process.supplier.templates.ui/src/net/openchrom/xxd/process/supplier/templates/ui/core/PeakReviewDirectMSD.java
@@ -39,6 +39,7 @@ import net.openchrom.xxd.process.supplier.templates.peaks.AbstractPeakIdentifier
 import net.openchrom.xxd.process.supplier.templates.settings.PeakReviewSettings;
 import net.openchrom.xxd.process.supplier.templates.ui.wizards.PeakReviewSupport;
 import net.openchrom.xxd.process.supplier.templates.ui.wizards.ProcessReviewSettings;
+import net.openchrom.xxd.process.supplier.templates.util.ChromatogramValidator;
 
 public class PeakReviewDirectMSD<T> extends AbstractPeakIdentifier implements IPeakIdentifierMSD<IIdentificationResults>, ITemplateExport {
 
@@ -51,12 +52,12 @@ public class PeakReviewDirectMSD<T> extends AbstractPeakIdentifier implements IP
 		if(peaks == null || peaks.isEmpty()) {
 			processingInfo.addErrorMessage(DESCRIPTION, "No peaks have been found in the current selection.");
 		} else {
-			runProcess(peaks, peakIdentifierSettings, processingInfo, monitor);
+			runProcess(peaks, processingInfo);
 		}
 		return processingInfo;
 	}
 
-	private void runProcess(List<? extends IPeakMSD> peaks, IPeakIdentifierSettingsMSD peakIdentifierSettings, IProcessingInfo<IIdentificationResults> processingInfo, IProgressMonitor monitor) {
+	private void runProcess(List<? extends IPeakMSD> peaks, IProcessingInfo<IIdentificationResults> processingInfo) {
 
 		/*
 		 * No settings: peakIdentifierSettings == null

--- a/openchrom/plugins/net.openchrom.xxd.process.supplier.templates.ui/src/net/openchrom/xxd/process/supplier/templates/ui/core/PeakReviewDirectMSD.java
+++ b/openchrom/plugins/net.openchrom.xxd.process.supplier.templates.ui/src/net/openchrom/xxd/process/supplier/templates/ui/core/PeakReviewDirectMSD.java
@@ -89,22 +89,27 @@ public class PeakReviewDirectMSD<T> extends AbstractPeakIdentifier implements IP
 		if(!reviewSettings.isEmpty()) {
 			PeakReviewSettings settings = new PeakReviewSettings();
 			settings.setReviewSettings(reviewSettings);
-			ProcessReviewSettings processSettings = new ProcessReviewSettings(processingInfo, chromatogram, settings);
-			try {
-				DisplayUtils.executeInUserInterfaceThread(new Runnable() {
+			List<ReviewSetting> filteredReviewSettings = ChromatogramValidator.filterValidReviewSettings(chromatogram, settings);
+			if(filteredReviewSettings.isEmpty()) {
+				processingInfo.addWarnMessage(DESCRIPTION, "The chromatogram doesn't contain any of the given peak traces.");
+			} else {
+				ProcessReviewSettings processSettings = new ProcessReviewSettings(processingInfo, chromatogram, settings);
+				try {
+					DisplayUtils.executeInUserInterfaceThread(new Runnable() {
 
-					@Override
-					public void run() {
+						@Override
+						public void run() {
 
-						Shell shell = DisplayUtils.getShell();
-						PeakReviewSupport peakReviewSupport = new PeakReviewSupport();
-						peakReviewSupport.addSettings(shell, processSettings);
-					}
-				});
-			} catch(InterruptedException e) {
-				Thread.currentThread().interrupt();
-			} catch(ExecutionException e) {
-				processingInfo.addErrorMessage(DESCRIPTION, "The execution failed, see attached log file.", e);
+							Shell shell = DisplayUtils.getShell();
+							PeakReviewSupport peakReviewSupport = new PeakReviewSupport();
+							peakReviewSupport.addSettings(shell, processSettings);
+						}
+					});
+				} catch(InterruptedException e) {
+					Thread.currentThread().interrupt();
+				} catch(ExecutionException e) {
+					processingInfo.addErrorMessage(DESCRIPTION, "The execution failed, see attached log file.", e);
+				}
 			}
 		}
 	}

--- a/openchrom/plugins/net.openchrom.xxd.process.supplier.templates.ui/src/net/openchrom/xxd/process/supplier/templates/ui/swt/peaks/ExtendedTargetsUI.java
+++ b/openchrom/plugins/net.openchrom.xxd.process.supplier.templates.ui/src/net/openchrom/xxd/process/supplier/templates/ui/swt/peaks/ExtendedTargetsUI.java
@@ -21,6 +21,7 @@ import org.eclipse.chemclipse.model.identifier.IIdentificationTarget;
 import org.eclipse.chemclipse.model.targets.ITarget;
 import org.eclipse.chemclipse.rcp.ui.icons.core.ApplicationImageFactory;
 import org.eclipse.chemclipse.rcp.ui.icons.core.IApplicationImage;
+import org.eclipse.chemclipse.rcp.ui.icons.core.IApplicationImageProvider;
 import org.eclipse.chemclipse.support.ui.events.IKeyEventProcessor;
 import org.eclipse.chemclipse.support.ui.menu.ITableMenuEntry;
 import org.eclipse.chemclipse.support.ui.swt.ExtendedTableViewer;
@@ -94,7 +95,7 @@ public class ExtendedTargetsUI extends Composite {
 		/*
 		 * Select the first entry if available.
 		 */
-		if(targets != null && targets.size() > 0) {
+		if(targets != null && !targets.isEmpty()) {
 			targetListUI.getTable().select(0);
 		}
 		//
@@ -150,9 +151,9 @@ public class ExtendedTargetsUI extends Composite {
 
 				boolean visible = PartSupport.toggleCompositeVisibility(searchSupportUI);
 				if(visible) {
-					button.setImage(ApplicationImageFactory.getInstance().getImage(IApplicationImage.IMAGE_SEARCH, IApplicationImage.SIZE_16x16));
+					button.setImage(ApplicationImageFactory.getInstance().getImage(IApplicationImage.IMAGE_SEARCH, IApplicationImageProvider.SIZE_16x16));
 				} else {
-					button.setImage(ApplicationImageFactory.getInstance().getImage(IApplicationImage.IMAGE_SEARCH, IApplicationImage.SIZE_16x16));
+					button.setImage(ApplicationImageFactory.getInstance().getImage(IApplicationImage.IMAGE_SEARCH, IApplicationImageProvider.SIZE_16x16));
 				}
 			}
 		});
@@ -165,7 +166,7 @@ public class ExtendedTargetsUI extends Composite {
 		Button button = new Button(parent, SWT.PUSH);
 		button.setText("");
 		button.setToolTipText("Open the Settings");
-		button.setImage(ApplicationImageFactory.getInstance().getImage(IApplicationImage.IMAGE_CONFIGURE, IApplicationImage.SIZE_16x16));
+		button.setImage(ApplicationImageFactory.getInstance().getImage(IApplicationImage.IMAGE_CONFIGURE, IApplicationImageProvider.SIZE_16x16));
 		button.addSelectionListener(new SelectionAdapter() {
 
 			@Override
@@ -197,8 +198,7 @@ public class ExtendedTargetsUI extends Composite {
 
 		Object object = targetListUI.getStructuredSelection().getFirstElement();
 		if(object instanceof IIdentificationTarget) {
-			IIdentificationTarget identificationTarget = (IIdentificationTarget)object;
-			return identificationTarget;
+			return (IIdentificationTarget)object;
 		}
 		return null;
 	}

--- a/openchrom/plugins/net.openchrom.xxd.process.supplier.templates.ui/src/net/openchrom/xxd/process/supplier/templates/ui/swt/peaks/ReviewController.java
+++ b/openchrom/plugins/net.openchrom.xxd.process.supplier.templates.ui/src/net/openchrom/xxd/process/supplier/templates/ui/swt/peaks/ReviewController.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2020, 2021 Lablicate GmbH.
+ * Copyright (c) 2020, 2022 Lablicate GmbH.
  * 
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
@@ -68,7 +68,7 @@ public class ReviewController {
 		//
 		if(processSettings != null) {
 			List<ReviewSetting> reviewSettings = processSettings.getReviewSettings();
-			if(reviewSettings.size() > 0) {
+			if(!reviewSettings.isEmpty()) {
 				update(reviewSettings.get(0));
 			}
 		}
@@ -108,7 +108,7 @@ public class ReviewController {
 	 */
 	public void update(List<IPeak> peaks) {
 
-		IPeak peak = peaks.size() > 0 ? peaks.get(0) : null;
+		IPeak peak = !peaks.isEmpty() ? peaks.get(0) : null;
 		updateChartPeaks(peaks);
 		updateExtendedTargetsUI(peak);
 	}

--- a/openchrom/plugins/net.openchrom.xxd.process.supplier.templates/src/net/openchrom/xxd/process/supplier/templates/util/AbstractTemplateListUtil.java
+++ b/openchrom/plugins/net.openchrom.xxd.process.supplier.templates/src/net/openchrom/xxd/process/supplier/templates/util/AbstractTemplateListUtil.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2018, 2020 Lablicate GmbH.
+ * Copyright (c) 2018, 2022 Lablicate GmbH.
  * 
  * All rights reserved.
  * This program and the accompanying materials are made available under the
@@ -88,8 +88,7 @@ public abstract class AbstractTemplateListUtil<T extends ITemplateValidator> imp
 	public List<String> importItems(File file) {
 
 		List<String> items = new ArrayList<>();
-		try {
-			BufferedReader bufferedReader = new BufferedReader(new FileReader(file));
+		try (BufferedReader bufferedReader = new BufferedReader(new FileReader(file))) {
 			String item;
 			while((item = bufferedReader.readLine()) != null) {
 				IStatus status = validator.validate(item);
@@ -97,11 +96,10 @@ public abstract class AbstractTemplateListUtil<T extends ITemplateValidator> imp
 					items.add(item);
 				}
 			}
-			bufferedReader.close();
-		} catch(FileNotFoundException e1) {
-			logger.warn(e1);
-		} catch(IOException e1) {
-			logger.warn(e1);
+		} catch(FileNotFoundException e) {
+			logger.warn(e);
+		} catch(IOException e) {
+			logger.warn(e);
 		}
 		//
 		return items;
@@ -125,8 +123,8 @@ public abstract class AbstractTemplateListUtil<T extends ITemplateValidator> imp
 	@Override
 	public List<String> getList(String preferenceEntry) {
 
-		List<String> values = new ArrayList<String>();
-		if(preferenceEntry != "") {
+		List<String> values = new ArrayList<>();
+		if(!"".equals(preferenceEntry)) {
 			String[] items = parseString(preferenceEntry);
 			if(items.length > 0) {
 				for(String item : items) {
@@ -150,7 +148,7 @@ public abstract class AbstractTemplateListUtil<T extends ITemplateValidator> imp
 
 	private List<String> getValues(String[] items) {
 
-		List<String> values = new ArrayList<String>();
+		List<String> values = new ArrayList<>();
 		if(items != null) {
 			int size = items.length;
 			for(int i = 0; i < size; i++) {

--- a/openchrom/plugins/net.openchrom.xxd.process.supplier.templates/src/net/openchrom/xxd/process/supplier/templates/util/ChromatogramValidator.java
+++ b/openchrom/plugins/net.openchrom.xxd.process.supplier.templates/src/net/openchrom/xxd/process/supplier/templates/util/ChromatogramValidator.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2021 Lablicate GmbH.
+ * Copyright (c) 2021, 2022 Lablicate GmbH.
  *
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
@@ -23,7 +23,9 @@ import org.eclipse.chemclipse.wsd.model.core.IScanWSD;
 import org.eclipse.chemclipse.wsd.model.xwc.IExtractedWavelengthSignal;
 
 import net.openchrom.xxd.process.supplier.templates.model.DetectorSetting;
+import net.openchrom.xxd.process.supplier.templates.model.ReviewSetting;
 import net.openchrom.xxd.process.supplier.templates.settings.PeakDetectorSettings;
+import net.openchrom.xxd.process.supplier.templates.settings.PeakReviewSettings;
 
 public class ChromatogramValidator {
 
@@ -37,6 +39,18 @@ public class ChromatogramValidator {
 		}
 		//
 		return detectorSettings;
+	}
+
+	public static List<ReviewSetting> filterValidReviewSettings(IChromatogram<?> chromatogram, PeakReviewSettings settings) {
+
+		List<ReviewSetting> reviewSettings = new ArrayList<>();
+		for(ReviewSetting reviewSetting : settings.getReviewSettingsList()) {
+			if(chromatogramContainsTraces(chromatogram, reviewSetting.getTraces())) {
+				reviewSettings.add(reviewSetting);
+			}
+		}
+		//
+		return reviewSettings;
 	}
 
 	public static boolean chromatogramContainsTraces(IChromatogram<?> chromatogram, String tracesSetting) {

--- a/openchrom/plugins/net.openchrom.xxd.process.supplier.templates/src/net/openchrom/xxd/process/supplier/templates/util/ChromatogramValidator.java
+++ b/openchrom/plugins/net.openchrom.xxd.process.supplier.templates/src/net/openchrom/xxd/process/supplier/templates/util/ChromatogramValidator.java
@@ -29,6 +29,11 @@ import net.openchrom.xxd.process.supplier.templates.settings.PeakReviewSettings;
 
 public class ChromatogramValidator {
 
+	private ChromatogramValidator() {
+
+		// only static access
+	}
+
 	public static List<DetectorSetting> filterValidDetectorSettings(IChromatogram<?> chromatogram, PeakDetectorSettings settings) {
 
 		List<DetectorSetting> detectorSettings = new ArrayList<>();

